src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+import './GoodsList.scss';
+
+const goodsFromServer: string[] = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+export const GoodsList = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [isReverseSort, setReverseSort] = useState(false);
+  const [isAlphabetSort, setAlphabetSort] = useState(false);
+  const [isLengthSort, setLengthSort] = useState(false);
+
+  const reverseSort = () => {
+    setReverseSort(!isReverseSort);
+  };
+
+  const alphabetSort = () => {
+    setAlphabetSort(!isAlphabetSort);
+  };
+
+  const lengthSort = () => {
+    setLengthSort(!isLengthSort);
+  };
+
+  const reset = () => {
+    setGoods(goodsFromServer);
+    setReverseSort(false);
+    setAlphabetSort(false);
+    setLengthSort(false);
+  };
+
+  const sortGoods = (goodsArray: string[]): string[] => {
+    if (isAlphabetSort) {
+      return goodsArray.sort((a, b) => a.localeCompare(b));
+    }
+
+    if (isLengthSort) {
+      return goodsArray.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReverseSort) {
+      return goodsArray.reverse();
+    }
+
+    return goodsArray;
+  };
+
+  const renderingGoods = sortGoods([...goods]);
```

Comment:
Can use `useMemo` here
```suggestion
  const renderingGoods = prepareGoods([...goods]);
```